File: regalia-moshi/src/test/java/com/commit451/regalia/moshi/RealmListJsonAdapterFactoryTests.java
package com.commit451.regalia.moshi;

import com.commit451.regalia.test.TestUtil;
import com.squareup.moshi.JsonAdapter;
import com.squareup.moshi.Moshi;

import org.junit.Assert;
import org.junit.Test;

public class RealmListJsonAdapterFactoryTests {

    private static Moshi moshi = new Moshi.Builder()
            .add(new RealmListJsonAdapterFactory())
            .build();

    @Test
    public void parseList() throws Exception {
        String json = TestUtil.getFileText("cat-with-list-of-friends.json");
        JsonAdapter<Cat> jsonAdapter = moshi.adapter(Cat.class);
        Cat cat = jsonAdapter.fromJson(json);
        Assert.assertNotNull(cat);
        Assert.assertNotNull(cat.name);
        Assert.assertNotNull(cat.friends);
    }
}

File: regalia-moshi/src/test/java/com/commit451/regalia/moshi/RealmListJsonAdapterFactoryTests.kt
package com.commit451.regalia.moshi

import com.commit451.regalia.test.TestUtil.getFileText
import com.squareup.moshi.Moshi
import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
import org.junit.Assert
import org.junit.Test

class RealmListJsonAdapterFactoryTests {
    @Test
    @Throws(Exception::class)
    fun parseList() {
        val json = getFileText("cat_and_friends.json")
        val jsonAdapter = moshi.adapter(Cat::class.java)
        val cat = jsonAdapter.fromJson(json)
        cat?.apply {
            Assert.assertNotNull(this)
            Assert.assertNotNull(name)
            Assert.assertNotNull(friends)
        }

    }

    companion object {
        private val moshi = Moshi.Builder()
                .add(KotlinJsonAdapterFactory())
                .add(RealmListJsonAdapterFactory())
                .build()
    }
}